test_suite/tests/json.rs

```diff
@@ -413,5 +418,25 @@ fn test_registry() {
 		]
 	});
 
-	assert_eq!(serde_json::to_value(registry).unwrap(), expected_json,);
+	let serialized_registry = serde_json::to_value(&registry).unwrap();
+	assert_eq!(serialized_registry, expected_json,);
+
+	let serialized_registry = serialized_registry.to_string();
+	let deserialized_registry: Registry = serde_json::from_str(string_to_static_str(serialized_registry)).unwrap();
+
+	// since type_table is not serialized, we just compare strings & types
+	// to test if deserialization works
+
+	let ser_vec = registry.strings();
+	let deser_vec = deserialized_registry.strings();
+	assert!(ser_vec.eq(deser_vec));
+
+	let ser_btree = registry.iter();
+	let deser_btree = deserialized_registry.iter();
+	assert!(ser_btree.eq(deser_btree));
+}
+
+// [WARN!!!!] Leaks the memory of s
+fn string_to_static_str(s: String) -> &'static str {
```

Comment:
We should really not merge this hack and instead wait until we converted all `&'static str` into `String`. We used `&'static str` because at first we did not intend to implement `scale::Decode`.